### PR TITLE
A handful of random mappings

### PR DESCRIPTION
**Pull request by modmuss50**
Im not usually one to create mappings, but I wanted something else to do for a bit.

### CODE REVIEW

File: mappings/net/minecraft/JsonSerializableType.mapping [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/class_5336 net/minecraft/JsonSerializableType
```
Comment (Juuxel) [inline mappings/net/minecraft/JsonSerializableType.mapping]: 1. Should be `net/minecraft/util/JsonSerializableType`
2. What's the difference between this and `JsonSerializable`?

File: mappings/net/minecraft/resource/ServerResourceManager.mapping [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		ARG 1 isDedicatedServer
+	METHOD method_29465 functionManager ()Lnet/minecraft/class_5349;
+	METHOD method_29466 reload (Ljava/util/List;ZILjava/util/concurrent/Executor;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;
+	METHOD method_29468 lootConditionManager ()Lnet/minecraft/class_4567;
+	METHOD method_29469 lootManager ()Lnet/minecraft/class_60;
+	METHOD method_29470 registryTagManager ()Lnet/minecraft/class_3505;
+	METHOD method_29471 recipeManager ()Lnet/minecraft/class_1863;
+	METHOD method_29472 commandManager ()Lnet/minecraft/class_2170;
+	METHOD method_29473 serverAdvancementLoader ()Lnet/minecraft/class_2989;
+	METHOD method_29474 resourceManager ()Lnet/minecraft/class_3300;
```
Comment (Juuxel) [inline mappings/net/minecraft/resource/ServerResourceManager.mapping]: Should these have `get` prefixes?

File: mappings/net/minecraft/world/World.mapping [hunk 3]
```diff
@@ -64,6 +71,8 @@ CLASS net/minecraft/class_1937 net/minecraft/world/World
 	METHOD method_26372 getCollisionHeightAt (Lnet/minecraft/class_2338;)D
 		ARG 1 pos
 	METHOD method_27982 isDebugWorld ()Z
+	METHOD method_27983 registryKey ()Lnet/minecraft/class_5321;
+	METHOD method_29287 dimensionRegistryKey ()Lnet/minecraft/class_5321;
```
Comment (Juuxel) [inline mappings/net/minecraft/world/World.mapping]: Same thing about `get` here.

File: mappings/net/minecraft/world/gen/GeneratorOptions.mapping [hunk 4]
```diff
@@ -32,3 +32,4 @@ CLASS net/minecraft/class_5285 net/minecraft/world/gen/GeneratorOptions
 	METHOD method_28036 withBonusChest ()Lnet/minecraft/class_5285;
 	METHOD method_28037 toggleGenerateStructures ()Lnet/minecraft/class_5285;
 	METHOD method_28038 toggleBonusChest ()Lnet/minecraft/class_5285;
+	METHOD method_28609 dimensionMap ()Ljava/util/LinkedHashMap;
```
Comment (Juuxel) [inline mappings/net/minecraft/world/gen/GeneratorOptions.mapping]: ...and here.

File: mappings/net/minecraft/server/MinecraftServer.mapping [hunk 5]
```diff
@@ -88,12 +91,19 @@ CLASS net/minecraft/server/MinecraftServer
 	METHOD method_24489 stopDebug ()Lnet/minecraft/class_3696;
 	METHOD method_24490 endMonitor (Lnet/minecraft/class_4758;)V
 		ARG 1 monitor
+	METHOD method_27050 savePath (Lnet/minecraft/class_5218;)Ljava/nio/file/Path;
```
Comment (Juuxel) [inline mappings/net/minecraft/server/MinecraftServer.mapping]: Does this save a path or get a save path? Getters should have `get`.

File: mappings/net/minecraft/server/MinecraftServer.mapping [hunk 6]
```diff
@@ -88,12 +91,19 @@ CLASS net/minecraft/server/MinecraftServer
...
 	METHOD method_27051 syncChunkWrites ()Z
 	METHOD method_27725 convertLevel (Lnet/minecraft/class_32$class_5143;)V
 	METHOD method_27727 getStructureManager ()Lnet/minecraft/class_3485;
+	METHOD method_27728 saveProperties ()Lnet/minecraft/class_5219;
```
Comment (Juuxel) [inline mappings/net/minecraft/server/MinecraftServer.mapping]: Same thing about saving here.

File: mappings/net/minecraft/world/SaveProperties.mapping [hunk 7]
```diff
@@ -31,3 +31,4 @@ CLASS net/minecraft/class_5219 net/minecraft/world/SaveProperties
 	METHOD method_27440 getFormatName (I)Ljava/lang/String;
 		ARG 1 id
 	METHOD method_27859 getMainWorldProperties ()Lnet/minecraft/class_5268;
+	METHOD method_28057 generatorOptions ()Lnet/minecraft/class_5285;
```
Comment (Juuxel) [inline mappings/net/minecraft/world/SaveProperties.mapping]: Missing `get` here as well.

File: mappings/net/minecraft/server/function/FunctionManager.mapping [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+	METHOD <init> (ILcom/mojang/brigadier/CommandDispatcher;)V
+		ARG 1 level
+		ARG 2 commandDispatcher
+	METHOD method_29447 functions ()Ljava/util/Map;
+	METHOD method_29450 readLines (Lnet/minecraft/class_3300;Lnet/minecraft/class_2960;)Ljava/util/List;
+		ARG 0 resourceManager
+		ARG 1 id
+	METHOD method_29456 get (Lnet/minecraft/class_2960;)Ljava/util/Optional;
+		ARG 1 id
+	METHOD method_29458 tags ()Lnet/minecraft/class_3503;
```
Comment (Juuxel) [inline mappings/net/minecraft/server/function/FunctionManager.mapping]: `functions` and `tags` should probably have `get` as well?

File: mappings/net/minecraft/JsonSerializableType.mapping [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
+	FIELD field_25204 jsonSerializer Lnet/minecraft/class_5335;
+	METHOD method_29312 jsonSerializer ()Lnet/minecraft/class_5335;
```
Comment (Juuxel) [inline mappings/net/minecraft/JsonSerializableType.mapping]: This one is also missing `get`.

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/resource/ServerResourceManager.mapping (hunk 2 above)
Comment (modmuss50) [reply mappings/net/minecraft/resource/ServerResourceManager.mapping]: Yes, I kinda forgot about that everwhere. Thats what I get for moving around projects with diffrent conventions lol

File: mappings/net/minecraft/JsonSerializableType.mapping (hunk 1 above)
Comment (modmuss50) [reply mappings/net/minecraft/JsonSerializableType.mapping]: Yes, should be in util.

Its an odd class, as its just holds an instance of JsonSerializable, wasnt too sure what to go with here.

File: mappings/net/minecraft/item/ItemUsageUtil.mapping [hunk 10]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_5328 net/minecraft/item/ItemUsageUtil
```
Comment (liach) [inline mappings/net/minecraft/item/ItemUsageUtil.mapping]: Maybe just `ItemUsage`

File: mappings/net/minecraft/server/function/FunctionManager.mapping [hunk 11]
```diff
@@ -0,0 +1,20 @@
+CLASS net/minecraft/class_5349 net/minecraft/server/function/FunctionManager
```
Comment (liach) [inline mappings/net/minecraft/server/function/FunctionManager.mapping]: More like a function loader, as we still have the old command function manager

File: mappings/net/minecraft/resource/ServerResourceManager.mapping [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+	FIELD field_25335 resourceManager Lnet/minecraft/class_3296;
+	FIELD field_25336 commandManager Lnet/minecraft/class_2170;
+	FIELD field_25337 recipeManager Lnet/minecraft/class_1863;
+	FIELD field_25338 registryTagManager Lnet/minecraft/class_3505;
+	FIELD field_25339 lootConditionManager Lnet/minecraft/class_4567;
+	FIELD field_25340 lootManager Lnet/minecraft/class_60;
+	FIELD field_25341 serverAdvancementLoader Lnet/minecraft/class_2989;
+	FIELD field_25342 functionManager Lnet/minecraft/class_5349;
+	METHOD <init> (ZI)V
+		ARG 1 isDedicatedServer
```
Comment (liach) [inline mappings/net/minecraft/resource/ServerResourceManager.mapping]:
```suggestion
		ARG 1 dedicated
```

File: mappings/net/minecraft/resource/ServerResourceManager.mapping [hunk 13]
```diff
@@ -0,0 +1,20 @@
+CLASS net/minecraft/class_5350 net/minecraft/resource/ServerResourceManager
+	FIELD field_25335 resourceManager Lnet/minecraft/class_3296;
+	FIELD field_25336 commandManager Lnet/minecraft/class_2170;
+	FIELD field_25337 recipeManager Lnet/minecraft/class_1863;
+	FIELD field_25338 registryTagManager Lnet/minecraft/class_3505;
+	FIELD field_25339 lootConditionManager Lnet/minecraft/class_4567;
+	FIELD field_25340 lootManager Lnet/minecraft/class_60;
+	FIELD field_25341 serverAdvancementLoader Lnet/minecraft/class_2989;
+	FIELD field_25342 functionManager Lnet/minecraft/class_5349;
```
Comment (liach) [inline mappings/net/minecraft/resource/ServerResourceManager.mapping]: more like `functionLoader`, re. advancement manager vs loader

File: mappings/net/minecraft/server/function/FunctionManager.mapping [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+		ARG 1 level
+		ARG 2 commandDispatcher
+	METHOD method_29447 getFunctions ()Ljava/util/Map;
+	METHOD method_29450 readLines (Lnet/minecraft/class_3300;Lnet/minecraft/class_2960;)Ljava/util/List;
+		ARG 0 resourceManager
+		ARG 1 id
+	METHOD method_29456 get (Lnet/minecraft/class_2960;)Ljava/util/Optional;
+		ARG 1 id
+	METHOD method_29458 getTags ()Lnet/minecraft/class_3503;
+	METHOD method_29459 getOrCreateTag (Lnet/minecraft/class_2960;)Lnet/minecraft/class_3494;
```
Comment (liach) [inline mappings/net/minecraft/server/function/FunctionManager.mapping]: Imo just `getTag` if it returns an empty tag for an nonexistent one.

Comment (modmuss50) [reply mappings/net/minecraft/server/function/FunctionManager.mapping]: Its very basic, just calls tags.getOrCreate so I think this is fine.

File: mappings/net/minecraft/resource/ServerResourceManager.mapping (hunk 12 above)
Comment (modmuss50) [reply mappings/net/minecraft/resource/ServerResourceManager.mapping]: I will also change this to match in CommandManager

**Review by Juuxel — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Pull request merged**